tests/gentropy/dataset/test_l2g_feature.py

```diff
@@ -803,3 +812,106 @@ def _setup(self: TestCommonVepFeatureLogic, spark: SparkSession) -> None:
             ),
             _schema=StudyLocus.get_schema(),
         )
+
+
+class TestCommonGeneCountFeatureLogic:
+    """Test the CommonGeneCountFeatureLogic methods."""
+
+    @pytest.mark.parametrize(
+        ("feature_name", "expected_data", "protein_coding_only"),
+        [
+            (
+                "geneCount",
+                [
+                    {"studyLocusId": "1", "geneId": "gene1", "geneCount": 3},
+                    {"studyLocusId": "1", "geneId": "gene2", "geneCount": 3},
+                    {"studyLocusId": "1", "geneId": "gene3", "geneCount": 3},
+                ],
+                False,  # Test case for all genes
+            ),
+            (
+                "geneCountProteinCoding",
+                [
+                    {
+                        "studyLocusId": "1",
+                        "geneId": "gene1",
+                        "geneCountProteinCoding": 2,
+                    },
+                    {
+                        "studyLocusId": "1",
+                        "geneId": "gene2",
+                        "geneCountProteinCoding": 2,
```

Comment:
Cool, in this case after we convert the matrix to a wide format and fill the NA's with 0s, studyLocusId 1 and gene3 will have a geneCountProteinCoding count of 0.